src/main/java/org/owasp/esapi/reference/DefaultSecurityConfiguration.java

```diff
@@ -1441,21 +1441,47 @@ public Boolean getBooleanProp(String propertyName) throws ConfigurationException
         try {
             return esapiPropertyManager.getBooleanProp(propertyName);
         } catch (ConfigurationException ex) {
-            String property = properties.getProperty( propertyName );
+            String property = properties.getProperty(propertyName);
             if ( property == null ) {
                 throw new ConfigurationException( "SecurityConfiguration for " + propertyName + " not found in ESAPI.properties");
             }
-            if ( property.equalsIgnoreCase("true") || property.equalsIgnoreCase("yes" ) ) {
+            if ( property.equalsIgnoreCase("true") || property.equalsIgnoreCase("yes") ) {
                 return true;
             }
-            if ( property.equalsIgnoreCase("false") || property.equalsIgnoreCase( "no" ) ) {
+            if ( property.equalsIgnoreCase("false") || property.equalsIgnoreCase("no") ) {
                 return false;
             }
             throw new ConfigurationException( "SecurityConfiguration for " + propertyName + " has incorrect " +
                     "type");
         }
     }
 
+    /**
+     * {@inheritDoc}
+     * Looks for property in three configuration files in following order:
+     * 1.) In file defined as org.owasp.esapi.opsteam system property
+     * 2.) In file defined as org.owasp.esapi.devteam system property
+     * 3.) In ESAPI.properties
+     */
+    @Override
+    public Boolean getBooleanProp(String propertyName, Boolean defaultValue) {
+        try {
+            return esapiPropertyManager.getBooleanProp(propertyName);
+        } catch (ConfigurationException ex) {
+            String property = properties.getProperty(propertyName);
+            if ( property == null ) {
+                return defaultValue;
+            }
+            if ( property.equalsIgnoreCase("true") || property.equalsIgnoreCase("yes") ) {
+                return true;
+            }
+            if ( property.equalsIgnoreCase("false") || property.equalsIgnoreCase("no") ) {
+                return false;
+            }
+            return defaultValue;
```

Comment:
Ditto comment in **EsapiPropertyManager.java** about throwing `ConfigurationException` here.